app/main/forms/brief_forms.py

```diff
@@ -0,0 +1,91 @@
+from flask_wtf import Form
+from wtforms import IntegerField, SelectMultipleField
+from wtforms.validators import NumberRange
+
+
+class BriefSearchForm(Form):
+    page = IntegerField(default=1, validators=(NumberRange(min=1),))
+    status = SelectMultipleField("Status", choices=(
+        ("live", "Live",),
+        ("closed", "Closed",)
+    ))
+    # lot choices expected to be set at runtime
+    lot = SelectMultipleField("Category")
+
+    def __init__(self, *args, **kwargs):
+        """
+            Accepts extra keyword arguments:
+             - `framework` - information on the target framework as returned by the api
+             - `data_api_client` - a data api client (should be able to remove the need for this arg at some point)
+        """
+        super(BriefSearchForm, self).__init__(*args, **kwargs)
+        try:
+            # popping this kwarg so we don't risk it getting fed to wtforms default implementation which might use it
+            # as a data field if there were a name collision
+            framework = kwargs.pop("framework")
+            self._framework_slug = framework["slug"]
+            self.lot.choices = tuple((lot["slug"], lot["name"],) for lot in framework["lots"] if lot["allowsBrief"])
+        except KeyError:
+            raise TypeError("Expected keyword argument 'framework' with framework information")
```

Comment:
Why a TypeError rather than the original KeyError? This doesn't seem to be a problem with object types?